torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -555,6 +554,8 @@ const std::vector<ProcessGroupAgent::FutureInfo> ProcessGroupAgent::
       it = futureTimeouts_.erase(it);
     }
   }
+  lock.unlock();
```

Comment:
Not sure about this pattern - we pass in a lock to `processTimedOutFutures`, which then unlocks it. The caller of processTimedOutFutures is not necessarily aware that the lock has been unlocked right? I think we can keep it as explicit as it was before, and not pass the lock, unless there is a reason to do so.

Reply:
Notice I used `std::move(lock)`, unique_lock is not copyable, but movable. We only have one lock instance at the same time in `pollTimedOutRPCs`.